src/Datadog.Trace/Sampling/RuleBasedSampler.cs

```diff
@@ -80,14 +72,14 @@ public void RegisterRule(ISamplingRule rule)
             _rules.Add(rule);
         }
 
-        private SamplingPriority GetSamplingPriority(Span span, float rate)
+        private SamplingPriority GetSamplingPriority(Span span, float rate, bool agentSampling)
         {
             var sample = ((span.TraceId * KnuthFactor) % TracerConstants.MaxTraceId) <= (rate * TracerConstants.MaxTraceId);
             var priority = SamplingPriority.AutoReject;
 
             if (sample)
             {
-                if (_tracingWithoutLimitsEnabled)
+                if (!agentSampling)
```

Comment:
If we invert this `if` condition, we can simplify as:
```csharp
if (sample)
{
	if (agentSampling)
	{
	    priority = SamplingPriority.AutoKeep;
	}
	else if (_limiter.Allowed(span))
	{
		priority = SamplingPriority.AutoKeep;
	}
}
```
which further simplifies to:
```csharp
if (sample && (agentSampling || _limiter.Allowed(span)))
{
	priority = SamplingPriority.AutoKeep;
}
```
and if we drop the local `priority`:
```csharp
var sample = ...

return sample && (agentSampling || _limiter.Allowed(span)) ?
		   SamplingPriority.AutoKeep :
		   SamplingPriority.AutoReject;
```

Reply:
Is it okay with you if we keep the first suggestion for readability?

Reply:
Eh, I'll change it